src/networking.c

```diff
@@ -412,6 +445,28 @@ cc_oci_network_create(const struct cc_oci_config *const config,
 	return false;
 }
 
+JsonObject *
+cc_oci_network_devices_to_json (const struct cc_oci_config *config)
+{
+        JsonObject *device = NULL;
+        guint i;
+        struct cc_oci_net_if_cfg *if_cfg = NULL;
+
+        for (i=0; i < g_slist_length(config->net.interfaces); i++) {
+                if_cfg = (struct cc_oci_net_if_cfg *)
+                        g_slist_nth_data(config->net.interfaces, i);
+
+                if (if_cfg->vf_based) {
+                        g_debug("interface %s is vf based vf: bdf %s driver %s",if_cfg->ifname, 
+                               if_cfg->bdf, if_cfg->device_driver);  
+                        device = json_object_new ();
+                        json_object_set_string_member (device, "bdf", if_cfg->bdf);
+                        json_object_set_string_member (device, "driver", if_cfg->device_driver);
```

Comment:
Can't the `device` be returned here (and the one at the end just be `return NULL`)?